src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -35,31 +35,48 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
 
 
         public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
-            var args = argSet.Values<IMember>();
+            var args = argSet.Arguments;
+            var constraintArgs = argSet.ListArgument?.Values;
+
             if (args.Count == 0) {
                 // TODO: report that at least one argument is required.
                 return declaringModule.Interpreter.UnknownType;
             }
 
-            var name = (args[0] as IPythonConstant)?.GetString();
+            var name = (args[0].Value as IPythonConstant)?.GetString();
             if (string.IsNullOrEmpty(name)) {
                 // TODO: report that type name is not a string.
                 return declaringModule.Interpreter.UnknownType;
             }
 
-            var constraints = args.Skip(1).Select(a => {
+            var constraints = constraintArgs?.Select(a => {
                 // Type constraints may be specified as type name strings.
                 var typeString = (a as IPythonConstant)?.GetString();
                 return !string.IsNullOrEmpty(typeString) ? argSet.Eval.GetTypeFromString(typeString) : a.GetPythonType();
-            }).ToArray();
+            }).ToArray() ?? Array.Empty<IPythonType>();
+
             if (constraints.Any(c => c.IsUnknown())) {
                 // TODO: report that some constraints could not be resolved.
             }
 
-            var docArgs = new[] { $"'{name}'" }.Concat(constraints.Select(c => c.IsUnknown() ? "?" : c.Name));
-            var documentation = CodeFormatter.FormatSequence("TypeVar", '(', docArgs);
-
+            var documentation = GetDocumentation(args, constraints);
             return new GenericTypeParameter(name, declaringModule, constraints, documentation, location);
         }
+
+        private static string GetDocumentation(IReadOnlyList<IArgument> args, IReadOnlyList<IPythonType> constraints) {
+            var name = (args[0].Value as IPythonConstant).GetString();
+            var keyWordArgs = new List<string>();
+
+            foreach (var arg in args.Skip(1)) {
+                if (arg.ValueIsDefault)
```

Comment:
Nit: we always use braces, even for single statement blocks.

Reply:
`var arg in args.Skip(1).Where(x => !x.ValueIsDefault)` will avoid comparison inside loop

Reply:
In fact, `keyWordArgs.Add(args.Skip(1).Where(x => !x.ValueIsDefault).Select(x => $"{x.Name}={(x.Value as IPythonConstant)?.Value}")`

Reply:
that's sort of hard to read but I'll make the change.

Reply:
We have lots of those :-) not too hard with some Linq experience. You can indent it as we do in other places.
```cs
keyWordArgs.Add(
    args
        .Skip(1)
        .Where(x => !x.ValueIsDefault)
        .Select(x => $"{x.Name}={(x.Value as IPythonConstant)?.Value}"
);
```